test/new-e2e/examples/vmenv_withos_test.go

```diff
@@ -8,21 +8,51 @@ package examples
 import (
 	"testing"
 
-	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
-	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2os"
-	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2params"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/test"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+	"github.com/DataDog/test-infra-definitions/components"
+	"github.com/DataDog/test-infra-definitions/components/vm"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2vm"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
 	"github.com/stretchr/testify/assert"
 )
 
-type vmSuiteEx1 struct {
-	e2e.Suite[e2e.VMEnv]
+type Remote vm.ERemote
+
+var _ components.Importable = &Remote{}
+
+func (r *Remote) GetClient(t *testing.T) (*client.VMClient, error) {
+	return client.NewVMClient(t, &utils.Connection{
+		Host: r.Host,
+		User: r.Username,
+	}, r.OSType)
+}
+
+type myVMEnv struct {
+	VM *Remote
 }
 
-func TestVMSuiteEx1(t *testing.T) {
-	e2e.Run(t, &vmSuiteEx1{}, e2e.EC2VMStackDef(ec2params.WithOS(ec2os.UbuntuOS)))
+type myVMSuite struct {
+	test.BaseSuite[myVMEnv]
 }
 
-func (v *vmSuiteEx1) TestItIsUbuntu() {
-	res := v.Env().VM.Execute("cat /etc/os-release")
+func TestMyVMSuite(t *testing.T) {
+	test.Run(t, &myVMSuite{}, test.WithTypedPulumiProvisioner[myVMEnv](func(ctx *pulumi.Context, env *myVMEnv) error {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
As this code can possibly help the Windows local vm requirement, how do you see implementing a test that has one environment, with a VM, where the provisioner changes based on some flag or environment variable ?  Would it have two providers, one for pulumi and one for the local vm, and call one or the other based on the variable ?